### ISSUE TITLE
Text with formatting and empty lines is not animated correctly

### ISSUE THREAD

**Issue by olshevski** [jcubic/jquery.terminal#890]
### Issue summary

When I try to echo (with typing animation) some text with any kind of formatting and with some empty line, the empty lines are skipped during the animation and appear only when the animation ends. See the attached video:

https://github.com/jcubic/jquery.terminal/assets/5606565/e802e88b-52b3-4fd6-811d-f71d22f868c4

Any text without formatting is always animated correctly.

### Expected behavior

Any text should animate empty lines correctly, not depending on the presence of the formatting syntax.

### Actual behavior

The presence of the formatting syntax breaks the animation of empty lines in a text.

### Steps to reproduce

Here is the code from the video:

```html
<script>
    $(function () {
        $('body').terminal({
            test: function() {
                this.echo("Hello\n\n[[;green;]world]", { typing: true, delay: 200})
            }
        }, {
            greetings: null,
        })
    })
</script>
```

And here is the codepen: https://codepen.io/olshevski/pen/zYMMpgy

### Browser and OS

The issue is reproduced on:
MacOS Ventura 13.3.1
Google Chrome 115.0.5790.114 (Official Build) (arm64)
Safari 16.4 (18615.1.26.110.1)

**Comment by jcubic** [comment]
Thanks for reporting the issue.

**Comment by jcubic** [comment]
The problem is with the prompt, it doesn't render empty lines:

```javascript
$('body').terminal({
}, {
    greetings: null,
    prompt: "Hello\n\n[[;green;]world]"
})
```

**Comment by jcubic** [comment]
Fixed on the devel branch, the problem was that for an unknown reason I added `filter(Boolean)` to lines, and empty lines got removed before rendering.

**Comment by jcubic** [comment]
Released in 2.37.0
